Sources/Baggage/Baggage.swift

```diff
@@ -0,0 +1,237 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift Baggage Context open source project
+//
+// Copyright (c) 2020 Moritz Lang and the Swift Baggage Context project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+// ==== ----------------------------------------------------------------------------------------------------------------
+// MARK: Baggage
+
+/// A `Baggage` is a heterogeneous storage type with value semantics for keyed values in a type-safe
+/// fashion.
+///
+/// Its values are uniquely identified via `BaggageKey`s (by type identity). These keys also dictate the type of
+/// value allowed for a specific key-value pair through their associated type `Value`.
+///
+/// ## Defining keys and accessing values
+/// Baggage keys are defined as types, most commonly case-less enums (as no actual instances are actually required)
+/// which conform to the `Baggage.Key` protocol:
+///
+///     private enum TestIDKey: BaggageKey {
+///       typealias Value = String
+///     }
+///
+/// While defining a key, one should also immediately declare an extension on `BaggageProtocol`,
+/// to allow convenient and discoverable ways to interact with the baggage item, the extension should take the form of:
+///
+///     extension BaggageProtocol {
```

Comment:
Note to always put the property extensions on the `BaggageProtocol`

Reply:
^ Another reason why having the protocol be the most concise name is a good thing. People will tend to use that one automatically without needing to read a note like this.

Reply:
Yeah fair point. I'll remove this protocol entirely, it's only purpose was more sugar for potential adopters  but let's not prioritze sugar I suppose.